### DISCUSSION TITLE
`ADR Suggestion` Protocol for creating an ADR

### DISCUSSION THREAD

**Discussion by andped10** [easyscience/.github#14] [Ideas]
To ensure agreement and adaptation of a new Architecture Design Record (ADR) we use the following protocol.

# Suggestion 

1. At the appropriate GitHub level (`Organization` or `Repo`) create a new discussion (`Idea` or `Poll` category)
2. Prefix the name the discussion `ADR Suggestion,` and use a name that describes what the ADR concerns.
    - `ADR Suggestion, Method naming convention`
3. Make sure that the other team members are aware that a new suggestion and invite for discussion.
4. When discussion is done the proposal is either accepted or closed

# Accepted
1. Created a new discussion under the `ADR` category.
2. Use an appropriate and descriptive name.
3. Description of the ADR discussion should be a resume of what has been accepted.
4. Add links to the original discussion.
5. Close the original discussion.

# Revision
1. To revise an ADR one need to start a new discussion (`Idea` or `Poll` category).
2. When a decision has been reached in the new discussion the ADR can be updated.
3. The old ADR description should be added as a comment for to ADR discussion itself to maintain a history of the changes

**Comment by damskii9992** [comment]
The 3rd point in Revision is an attempt to have a version history of the ADR?

**Reply by andped10** [reply]
yes

**Comment by AndrewSazonov** [comment]
I reopened this because I’d like to suggest one minor change to this part:

> 2. Prefix the name the discussion `ADR Suggestion,` and use a name that describes what the ADR concerns.
    - `ADR Suggestion, Method naming convention`
    
I think it would be clearer if we used a code block, square brackets, or at least a colon to better separate the `ADR Suggestion` prefix from the rest of the title. The current comma-based format can be a bit confusing, since a comma often implies a list rather than a prefix.

Here’s a quick comparison of different options:

- ADR Suggestion, Unified Labeling System for EasyScience GitHub Organization  (current)
- ADR Suggestion: Unified Labeling System for EasyScience GitHub Organization  
- [ADR Suggestion] Unified Labeling System for EasyScience GitHub Organization  
- `ADR Suggestion` Unified Labeling System for EasyScience GitHub Organization  (suggested)

Personally, I’d lean toward using a code block (`ADR Suggestion`), because it looks more like a label rather than part of the title.
